sources/packages/web/src/components/common/Reports.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <full-page-container>
+    <template #header>
+      <header-navigator :title="title" subTitle="Reports" />
+    </template>
+    <body-header title="Export reports" />
+    <formio
+      formName="exportfinancialreports"
+      @loaded="formLoaded"
+      @submitted="exportReport"
+    ></formio>
+    <v-row class="justify-center m-4">
+      <check-permission-role :role="Role.AESTReports">
+        <template #="{ notAllowed }">
+          <v-btn color="primary" @click="submitForm" :disabled="notAllowed"
+            >Export CSV file</v-btn
+          >
+        </template>
+      </check-permission-role></v-row
+    >
+  </full-page-container>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { ReportsFilterAPIInDTO } from "@/services/http/dto";
+import {
+  useSnackBar,
+  useFileUtils,
+  useFormioDropdownLoader,
+} from "@/composables";
+import { FormIOForm, Role } from "@/types";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  components: {
+    CheckPermissionRole,
+  },
+  props: {
+    title: { type: String, required: true },
+    reportType: { type: Number, required: true },
```

Comment:
This component can receive the list of `OptionItemAPIOutDTO` that is expected to be loaded by each consumer (Ministry or institution).